nix/nim-status/getFlags.nix

```diff
@@ -0,0 +1,119 @@
+
+  targetArchMap = rec {
+    "386" = "i686";
+    "arm" = "arm";
+    "arm64" = "aarch64";
+  };
+  targetArch = lib.getAttr arch targetArchMap;
+  iosSdk = if arch == "386" then "iphonesimulator" else "iphoneos";
+  androidTarget = "${if arch == "arm" then "armv7a" else targetArch}-linux-${platform}";
+
+  ldArchMap = {
+      "386" = "x86";
+      "arm" = "arm";
+      "arm64" = "arm64";
+  };
+
+
+  ldArch = lib.getAttr arch ldArchMap;
+  androidToolPath = "${ANDROID_NDK_HOME}/toolchains/llvm/prebuilt/${osId}-${osArch}/bin";
+  androidToolPathPrefix = "${androidToolPath}/${targetArch}-linux-${platform}";
+  isysroot = if isAndroid then 
+    "${ANDROID_NDK_HOME}/toolchains/llvm/prebuilt/${osId}-${osArch}/sysroot"             
+    else "$(xcrun --sdk ${iosSdk} --show-sdk-path)";
+
+  compilerFlags = if isAndroid then
+      "-isysroot ${isysroot} -target ${androidTarget}${api} -fPIC"
+      else if isIOS then
+      # TODO The conditional for -miphoneos-version-min=8.0 is required,
+      # otherwise Nim will complain that thread-local storage is not supported for the current target
+      # when expanding 'NIM_THREADVAR' macro
+
+      "-isysroot ${isysroot} -fembed-bitcode -arch ${iosArch} ${if fromNim && arch == "arm" then "" else "-m${iosSdk}-version-min=8.0"}"
+      else throw "Unsupported platform!";
+
+  linkerFlags = if isAndroid then  
+  "--sysroot ${isysroot} -target ${androidTarget}${api}"
+  else if isIOS then
+  "--sysroot ${isysroot} -fembed-bitcode -arch ${iosArch} ${if fromNim && arch == "arm" then "" else "-m${iosSdk}-version-min=8.0"}"
+  else throw "Unsupported platform!";
+
+
+  iosToolPath = "${xcodeWrapper}/bin";
+
+  compilerVars = if isAndroid then
+    ''
+      export PATH=${androidToolPath}:$PATH
+      export AR=${androidToolPathPrefix}-ar
+      export RANLIB=${androidToolPathPrefix}-ranlib
+      # go build will use this
+      export CC=${androidToolPath}/clang
+
+      # This is important, otherwise Nim might not use proper tooling
+      mkdir bin
+      ln -s $AR bin/ar
+      ln -s $RANLIB bin/ranlib
+      export PATH=`pwd`/bin:$PATH
+    ''
+    else if isIOS then
+    ''
+      export PATH=${iosToolPath}:$PATH
+    ''
+    else throw "Unsupported platform!";
+
+  toolPath = if isAndroid then androidToolPath else iosToolPath;
+  hostMap = {
+    "386" = "x86";
+    "arm" = "arm";
+    "arm64" = "aarch64";
+  };
+  hostFlag = if isAndroid then androidTarget else lib.getAttr arch hostMap;
+
+  # Arg arch -> Nim arch
+  nimCpuMap = {
+    "386" = "i386";
+    "x86_64" = "amd64"; 
+    "arm" = "arm"; 
+    "arm64" = "arm64";
+  };
+
+  nimCpu = if arch=="386" && platform == "ios" then "amd64" else "${lib.getAttr arch nimCpuMap}";
+  nimPlatform = "${(if platform == "ios" then "ios" else "android")}";
+
+in {
+  "compiler" = compilerFlags;
+  "linker" = linkerFlags;
+  "vars" = compilerVars;
+  "host" = hostFlag;
+  "isysroot" = isysroot;
```

Comment:
Unused? As is `isAndroid` and `isIOS` as far as I can tell.

Reply:
Removed.